base/int.jl

```diff
@@ -552,10 +559,57 @@ end
 
 # checked ops are broken for 128-bit types (LLVM bug) ## FIXME: #4905
 
-checked_add(x::Int128, y::Int128) = x + y
-checked_sub(x::Int128, y::Int128) = x - y
-checked_mul(x::Int128, y::Int128) = x * y
+function checked_add(x::Int128, y::Int128)
+    # # x + y > typemax(Int128)
+    # y >= 0 && x > typemax(Int128) - y && throw(OverflowError())
+    # # x + y < typemin(Int128)
+    # y < 0 && x < typemin(Int128) - y && throw(OverflowError())
+    # x + y
+    r = x + y
+    # x and y have the same sign, and the result has a different sign
+    x$y >= 0 && x$r < 0 && throw(OverflowError())
```

Comment:
i'm assuming, you added parentheses when you changed `&&` to `&`? otherwise the precedence rules would rewrite that expression to `x$y >= 0 < 0` => false and should have deleted the entire thing